components/base/kinematicbase/kinematics.go

```diff
@@ -102,50 +107,91 @@ func (ddk *differentialDriveKinematics) CurrentInputs(ctx context.Context) ([]re
 func (ddk *differentialDriveKinematics) GoToInputs(ctx context.Context, desired []referenceframe.Input) (err error) {
 	// create capsule which defines the valid region for a base to be when driving to desired waypoint
 	// deviationThreshold defines max distance base can be from path without error being thrown
-	current, err := ddk.CurrentInputs(ctx)
-	if err != nil {
-		return err
+	current, inputsErr := ddk.CurrentInputs(ctx)
+	if inputsErr != nil {
+		return inputsErr
 	}
-	validRegion, err := ddk.newValidRegionCapsule(current, desired)
-	if err != nil {
-		return err
+	validRegion, capsuleErr := ddk.newValidRegionCapsule(current, desired)
+	if capsuleErr != nil {
+		return capsuleErr
 	}
+	movementErr := make(chan error, 1)
+	defer close(movementErr)
 
-	// this loop polls the error state and issues a corresponding command to move the base to the objective
-	// when the base is within the positional threshold of the goal, exit the loop
-	for err = ctx.Err(); err == nil; err = ctx.Err() {
-		col, err := validRegion.CollidesWith(spatialmath.NewPoint(r3.Vector{X: current[0].Value, Y: current[1].Value}, ""))
-		if err != nil {
-			return err
-		}
-		if !col {
-			return errors.New("base has deviated too far from path")
-		}
+	cancelContext, cancel := context.WithCancel(ctx)
```

Comment:
chose to use cancel context after discussing w @biotinker

Reply:
A cancellable context is what allows us to actually cease the movement when we detect and return an error.

This should have a `defer cancel()` so that any exit of this function will cause the spawned goroutines to also cancel.